packages/Parallel/Application/ParallelFileProcessor.php

```diff
@@ -195,12 +214,20 @@ function (array $json) use (
                         $this->processPool->quitAll();
                     }
 
+                    if ($fileBudgetPerProcess[$processIdentifier] <= 0) {
+                        // kill the current worker, and spawn a fresh one to free memory
+                        $this->processPool->quitProcess($processIdentifier);
+
+                        ($processSpawner)();
+                        return;
+                    }
                     if ($jobs === []) {
                         $this->processPool->quitProcess($processIdentifier);
                         return;
                     }
 
                     $job = array_pop($jobs);
+                    $fileBudgetPerProcess[$processIdentifier]--;
```

Comment:
Maybe I should move this one line below

Reply:
You should use `--$fileBudgetPerProcess[$processIdentifier]`

```suggestion
                    --$fileBudgetPerProcess[$processIdentifier];
```